File: VoiceRecorder/Scales.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using VoiceRecorder.Audio;

namespace VoiceRecorder
{
    static class PredefinedScales
    {
        public static readonly HashSet<Note> Chromatic = new HashSet<Note>() { Note.C, Note.CSharp, Note.D, Note.DSharp, Note.E, Note.F, Note.FSharp, Note.G, Note.GSharp, Note.A, Note.ASharp, Note.B };

        private static HashSet<Note> MakeScale(Note start, IEnumerable<int> offsets)
        {
            HashSet<Note> scale = new HashSet<Note>();
            foreach (int n in offsets)
            {
                scale.Add((Note)(((int)start + n) % 12));
            }
            return scale;
        }

        public static HashSet<Note> MakeMajorScale(Note start)
        {
            return MakeScale(start, new[] { 0, 2, 4, 5, 7, 9, 11 });
        }

        public static HashSet<Note> MakePentatonicScale(Note start)
        {
            return MakeScale(start, new[] { 0, 2, 4, 7, 9 });
        }
    }
}

File: ~VB Port/VoiceRecorder/Scales.vb
Imports System.Text
Imports VoiceRecorder.Audio

Namespace VoiceRecorder
    Friend NotInheritable Class PredefinedScales
        Public Shared ReadOnly Chromatic As New HashSet(Of Note) From {Note.C,
                                                                       Note.CSharp,
                                                                       Note.D,
                                                                       Note.DSharp,
                                                                       Note.E,
                                                                       Note.F,
                                                                       Note.FSharp,
                                                                       Note.G,
                                                                       Note.GSharp,
                                                                       Note.A,
                                                                       Note.ASharp,
                                                                       Note.B}

        Private Sub New()
        End Sub

        Private Shared Function MakeScale(ByVal start As Note, ByVal offsets As IEnumerable(Of Integer)) As HashSet(Of Note)
            Dim scale As New HashSet(Of Note)
            For Each n In offsets
                scale.Add(CType((CInt(Fix(start)) + n) Mod 12, Note))
            Next n
            Return scale
        End Function

        Public Shared Function MakeMajorScale(ByVal start As Note) As HashSet(Of Note)
            Return MakeScale(start, { 0, 2, 4, 5, 7, 9, 11 })
        End Function

        Public Shared Function MakePentatonicScale(ByVal start As Note) As HashSet(Of Note)
            Return MakeScale(start, { 0, 2, 4, 7, 9 })
        End Function
    End Class
End Namespace
